code/modules/hydroponics/rancher/incubator.dm

```diff
@@ -0,0 +1,46 @@
+/obj/machinery/incubator
+	name = "egg incubator"
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "incubator"
+	density = 1
+	anchored = 0
+	var/obj/item/weapon/reagent_containers/food/snacks/egg/incubated = null
+
+/obj/machinery/incubator/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/incubator(null)
+	component_parts += new /obj/item/weapon/stock_parts/manipulator(null)
+	RefreshParts()
+
+/obj/machinery/incubator/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/egg))
+		user.drop_item(O)
+		O.loc = src
+		user << "You add [O] to [src]."
+		icon_state = "incubator_e"
+		incubated = O
+		SSobj.processing |= incubated
+		return
+	if(default_deconstruction_screwdriver(user, "incubator", "incubator", O))
```

Comment:
Add a fucking grew square indicating the open hatch goddam is it really hard to do?